packages/wrangler/src/index.tsx

```diff
@@ -1485,8 +1500,11 @@ export async function main(argv: string[]): Promise<void> {
               // @ts-expect-error non-standard property on Error
               if (e.code === 10007) {
                 // upload a draft worker
+                // TODO: log a warning
                 await fetchResult(
-                  `/accounts/${config.account_id}/workers/scripts/${scriptName}`,
+                  !args["legacy-env"] && args.env
```

Comment:
filed https://github.com/cloudflare/wrangler2/issues/524 to write tests for this codepath, looks like I missed it last time around